apps/protocol-frontend/src/hooks/useContributionCount.ts

```diff
@@ -0,0 +1,62 @@
+import { useQuery } from '@tanstack/react-query';
+import { useUser } from '../contexts/UserContext';
+import { subWeeks } from 'date-fns';
+
+const useContributionCountInRange = (args: {
+  startDate: Date | string;
+  endDate: Date | string;
+  guildIds?: number[] | null | undefined;
+  excludeUnassigned?: boolean;
+}) => {
+  const { userData, govrnProtocol: govrn } = useUser();
+
+  const { isLoading, isFetching, isError, error, data } = useQuery(
+    ['GetContributionCount', args],
+    async () => {
+      if (!userData?.id) {
+        console.error('getUserContributionsCount has no userData.id');
+        return [];
+      }
+
+      return await govrn.custom.getContributionCountByDateForUserInRange({
+        id: userData?.id,
+        startDate: args.startDate,
+        endDate: args.endDate,
+        guildIds: args.guildIds,
+        excludeUnassigned: args.excludeUnassigned,
+      });
+    },
+  );
+
+  return { isLoading, isError, isFetching, error, data };
+};
+
+export const useContributionCountInYear = (args: {
```

Comment:
I think these can be combined. The param for start date can be passed in as an argument.

Reply:
Did you mean the two hooks?

Reply:
Yes

Reply:
The two requested will be associated with the same key; They both will have the same response.

Reply:
They shouldn't have the same key because they should have different arguments

Reply:
Yeah. It's working! 5568cfb3b5b6bf74769c141b997c14b37e348479